libs/openchallenges/challenge/src/lib/challenge.component.ts

```diff
@@ -22,46 +29,47 @@ import { SeoService } from '@sagebionetworks/shared/util';
 })
 export class ChallengeComponent implements OnInit {
   public appVersion: string;
-  account$!: Observable<Account | undefined>;
   challenge$!: Observable<Challenge>;
   loggedIn = false;
   progressValue = 0;
   remainDays!: number | undefined;
   challengeAvatar!: Avatar;
   tabs = CHALLENGE_TABS;
   tabKeys: string[] = Object.keys(this.tabs);
-  activeTab: Tab = this.tabs['overview'];
+  activeTab!: Tab;
   private subscriptions: Subscription[] = [];
 
   constructor(
     private activatedRoute: ActivatedRoute,
     private router: Router,
-    private readonly configService: ConfigService,
-    private seoService: SeoService,
-    private renderer2: Renderer2
+    private challengeService: ChallengeService,
+    private readonly configService: ConfigService
   ) {
     this.appVersion = this.configService.config.appVersion;
   }
 
   ngOnInit(): void {
-    this.activatedRoute.params.subscribe((param) => {
-      const challenge = DEPRECATED_MOCK_CHALLENGES.find(
-        (c) => c.name === param['challengeName']
-      );
-      if (challenge) {
-        this.challenge$ = of(challenge);
-      }
-    });
-
-    const activeTab$ = this.activatedRoute.queryParamMap.pipe(
-      map((params: ParamMap) => params.get('tab')),
-      map((key) => (key === null ? 'overview' : key))
+    this.challenge$ = this.activatedRoute.params.pipe(
+      switchMap((params) =>
+        this.challengeService.getChallenge(params['challengeId'])
+      ),
+      switchMap((challenge) => {
+        this.router.navigate(['/challenge', challenge.id, challenge.slug]);
+        return of(challenge);
+      }),
+      catchError((err) => {
+        const errorMsg = handleHttpError(err, this.router, {
```

Comment:
The variable name `errorMsg` is ill-defined: I would expect an error message based on the name (string or object) but this is actually a function, so the variable name should be a verb and here be more specific, e.g. `handleError`.

Reply:
@tschaffter Thanks for the advice. The `handleHttpError` function is used to 
- validate if a Error object is ApiClientError (that has status and title in the object)
- redirect to the page based on http status code
- return a new error (message), which will be emitted in the console.
```ts
export function handleHttpError(
  error: any,
  router: Router,
  httpStatusRedirect: HttpStatusRedirect,
  defaultRedirectUrl = '/'
): Error {
  const err = error.error;
  const status = error.status as number;
  const message = isApiClientError(err) ? err.detail : error.message;
  // redirect to corresponding url based on http status code
  // if no url is specified for the status code, redirect to default url
  router.navigate([httpStatusRedirect[status] || defaultRedirectUrl]);
  // return the error message
  return new Error(message);
}

```

Reply:
I see. Then maybe rename the variable `const error` instead of `const errorMsg`.